### ISSUE TITLE
How to power and connect HC-SR04 sonar sensors

### ISSUE THREAD

**Issue by nagen1** [lukas/robot#7]
@lukas ,
was trying to connect 3 HC-SR04 same like you did and wondering how can I connect them.
in your instruction, you mentioned which GPIO ports we can use but wondering how you powered then !? did you used Raspberry pi on-board 5v for 3 sensors!? OR powered then with externally battery back !? 

Any in-site and information much appreciated.

Regards
Nagen

**Comment by lukas** [comment]
You can use the GPIO 5v pin.

**Comment by nagen1** [comment]
I am beginner.. please don't mind, can I connect 3 sensors to one power source as I am already using it to connect/run my robot motors !? 5v on board out!

**Comment by lukas** [comment]
Yeah, that should work.  Did you try it?

**Comment by nagen1** [comment]
didn't tried it yet as I was facing RaspberryPI freeze issue while running using sensor. it's wired though and finally last night was able to get response from sensor. I will connect all of them to breadboard and will see how it works. Will keep you posted.

just asking - you can see my repo and correct me if you know why freeze issue is happening :)
https://github.com/nagen1/robot-4wd

**Comment by nagen1** [comment]
Hi Lukas,
I tried connecting 3 sensors to one RPi 5v and ground using breadboard it works awesome... :)
I typically took your details in Oreilly article and improved a little bit. my entire hardware is now running on one battery pack 10000Amh (the one you suggested) and everything works great.

here is how I connected all of them.
- 3 sensors connected to breadboard, 5v powers all sensors and ground
- motors power connected using USB out from battery pack (got ride of 6AA batteries pack)
- RPi gets power from same USB battery pack
